tests/python/unittest/test_while_loop.py

```diff
@@ -0,0 +1,969 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import mxnet as mx
+from mxnet import gluon
+import numpy as np
+import copy
+from numpy.testing import assert_allclose
+import unittest
+from mxnet.test_utils import almost_equal, default_context
+from numpy.testing import assert_allclose as assert_almost_equal  # This is more restrictive
+from mxnet.base import _as_list
+
+
+def test_while_loop_simple_forward():
+
+    class _TestBlock(gluon.HybridBlock):
+
+        def __init__(self, cond, func, max_iterations):
+            super(_TestBlock, self).__init__()
+            self.cond = cond
+            self.func = func
+            self.max_iterations = max_iterations
+
+        def hybrid_forward(self, F, *loop_vars):
+            return F.contrib.while_loop(
+                cond=self.cond,
+                func=self.func,
+                loop_vars=loop_vars,
+                max_iterations=self.max_iterations
+            )
+
+    for hybridize in [False, True]:
+        # Case 1.1: result should be sum([1, 2, 3 ... 100])
+        model = _TestBlock(
+            cond=lambda i, s: i <= 5,
+            func=lambda i, s: (None, (i + 1, s + i)),
+            max_iterations=10,
+        )
+        if hybridize:
+            model.hybridize(inline_limit=0)
```

Comment:
is `inline_limit` necessary?

Reply:
i don't think people use `inline_limit` by default. we should test common cases.

Reply:
Removed